src/object_store/src/object/mem.rs

```diff
@@ -130,23 +131,15 @@ impl ObjectStore for InMemObjectStore {
         }))
     }
 
-    async fn read(&self, path: &str, block: Option<BlockLocation>) -> ObjectResult<Bytes> {
+    async fn read(
+        &self,
+        path: &str,
+        range: impl RangeBounds<usize> + Clone + Send + Sync + std::fmt::Debug + 'static,
```

Comment:
May define `pub trait ObjectStoreRange = RangeBounds<usize> + Clone + Send + Sync + std::fmt::Debug + 'static` to avoid defining the long trait for multiple times.